src/test/Serialize_test.cpp

```diff
@@ -0,0 +1,224 @@
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <Serialize.h>
+#include <RippleKey.h>
+#include <test/KnownTestData.h>
+#include <test/KeyFileGuard.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/HashPrefix.h>
+#include <ripple/protocol/Sign.h>
+#include <ripple/protocol/TxFlags.h>
+#include <beast/core/detail/base64.hpp>
+
+namespace serialize {
+
+namespace test {
+
+class Serialize_test : public beast::unit_test::suite
+{
+private:
+    void verifyKnownTx(ripple::STTx const& obj)
+    {
+        using namespace ripple;
+        BEAST_EXPECT(to_string(obj.getTransactionID()) ==
+            "F2D008D2AABBABD2A882F9049AA873210908EC3EA1EB0A2044A66093C7ACD2B1");
+        BEAST_EXPECT(obj[sfTransactionType] == ttPAYMENT);
+        BEAST_EXPECT(obj[sfFlags] == tfFullyCanonicalSig);
+        BEAST_EXPECT(toBase58(obj[sfAccount]) ==
+            "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn");
+        BEAST_EXPECT(obj[sfSequence] == 18);
+        BEAST_EXPECT(obj[sfFee] == STAmount(100));
+        BEAST_EXPECT(strHex(obj[sfSigningPubKey]) ==
+            "0388935426E0D08083314842EDFBB2D517BD47699F9A4527318A8E10468C97C052");
+        BEAST_EXPECT(strHex(obj[sfTxnSignature]) ==
+            "3044022030425DB6A46B5B57BDA85E5B8455B90DC4EC57BA1A707AF0C"
+            "28DC9383E09643D0220195B9FDBE383B813A539F3B70E130482E92D1E"
+            "1210B0F85551E11B3F81EB98BB");
+        BEAST_EXPECT(toBase58(obj[sfDestination]) ==
+            "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh");
+        auto const amountAccount = parseBase58<AccountID>(
+            "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq");
+        if (BEAST_EXPECT(amountAccount))
+        {
+            BEAST_EXPECT(obj[sfAmount] == STAmount(
+                Issue(to_currency("USD"), *amountAccount),
+                    123400000));
+        }
+        auto const sendMaxAccount = parseBase58<AccountID>(
+            "razqQKzJRdB4UxFPWf5NEpEG3WMkmwgcXA");
+        if (BEAST_EXPECT(sendMaxAccount))
+        {
+            // Need a variable here instead of a literal to make
+            // some compilers happy
+            const std::uint64_t mantissa = 5678900000000000LLU;
+            BEAST_EXPECT(obj[sfSendMax] == STAmount(
+                Issue(to_currency("CNY"), *sendMaxAccount),
+                mantissa, -4));
+        }
+    }
+
+    void testParseJson()
+    {
+        testcase("ParseJson");
+
+        auto& testTx = getKnownTx();
+        auto json = parseJson(testTx.JsonText);
+        BEAST_EXPECT(json);
+
+        // Don't need to test the `JsonReader`, just a case to ensure the wrapper
+        // handles failures
+        json = parseJson("{ asjlfkjs");
+        BEAST_EXPECT(!json);
+    }
+
+    void testMakeObject()
+    {
+        testcase("Make Object");
+
+        auto& testTx = getKnownTx();
```

Comment:
The compiler seems to automagically do the right thing for these references, making them `const`, but to clarify intent, I'll make that explicit.